Library/Homebrew/cmd/update.sh

```diff
@@ -352,7 +352,10 @@ homebrew-update() {
         HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH=1
         ;;
       --auto-update) export HOMEBREW_UPDATE_AUTO=1 ;;
-      --*) ;;
+      --*)
+        brew help update
+        exit $?
```

Comment:
I think this should just `exit 1` with unknown options, and also output the unknown option.